lib/active_support/cache/database_store.rb

```diff
@@ -21,6 +28,7 @@ def self.supports_cache_versioning?
       # option options [Class] :model model class. Default: ActiveSupport::Cache::DatabaseStore::Model
       def initialize(options = nil)
         @model = (options || {}).delete(:model) || Model
+        @compression = (options || {}).delete(:compression) || 'plain'
```

Comment:
```rb
@compression = (options || {}).delete(:compression)&.to_s
```

in case someone passes a symbol + don't bother with `'plain'`. please also add:

```rb
raise AgumentError, "invalid compression option #{@compression.inspect}" if @compression && !COMPRESSION_HANDLERS.key?(@compression)
```

Reply:
you need to document the option